File: Server/models/AISession.js
const mongoose = require('mongoose');

const aiSessionSchema = new mongoose.Schema({
  session_id: {
    type: String,
    required: true,
    unique: true,
    index: true
  },
  user_id: {
    type: Number,
    required: true,
    ref: 'User',
    index: true
  },
  title: {
    type: String,
    default: null
  }
}, {
  timestamps: { createdAt: 'created_at', updatedAt: 'updated_at' } 
});

// tối ưu tốc độ load Sidebar trên App
aiSessionSchema.index({ updated_at: -1 });

module.exports = mongoose.model('AISession', aiSessionSchema, 'ai_sessions');

File: Android/app/src/main/java/com/example/smartfashion/model/AiSession.kt
package com.example.smartfashion.model

import com.google.gson.annotations.SerializedName
import java.util.Date

data class AiSession(
    @SerializedName("_id")
    val id: String? = null,

    @SerializedName("session_id")
    val sessionId: String,

    @SerializedName("user_id")
    val userId: Int,

    @SerializedName("title")
    val title: String? = null,

    @SerializedName("created_at")
    val createdAt: Date? = null,

    @SerializedName("updated_at")
    val updatedAt: Date? = null
)